wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -50,14 +51,37 @@ import Network.Wai.Middleware.RequestLogger.Internal
 import Network.Wai.Header (contentLength)
 import Data.Text.Encoding (decodeUtf8')
 
+-- | The logging format.
 data OutputFormat
   = Apache IPAddrSource
-  | Detailed Bool -- ^ use colors?
+  | Detailed DetailedSettings
   | CustomOutputFormat OutputFormatter
   | CustomOutputFormatWithDetails OutputFormatterWithDetails
   | CustomOutputFormatWithDetailsAndHeaders OutputFormatterWithDetailsAndHeaders
 
-type OutputFormatter = ZonedDate -> Request -> Status -> Maybe Integer -> LogStr
+-- | Settings for the `Detailed` `OutputFormat`.
+--
+-- `paramFilter` allows you to pass a function to hide confidential
+-- information (such as passwords) from the logs. If the second parameter is `Just`, then POST
+-- bodies are also hidden. For example:
+-- > myformat = Detailed True (Just hidePasswords)
+-- >   where hidePasswords p@(k,v) = if k = "password" then (k, "***REDACTED***") else p
+--
+-- `requestFilter` allows you to filter which requests are logged.
+data DetailedSettings = DetailedSettings
+    { useColors :: Bool
+    , mModifyParams :: Maybe (Param -> Param)
```

Comment:
Shouldn't this be 
```suggestion
    , mModifyParams :: Maybe (Param -> Maybe Param)
```
?
That will also allow one to _remove_ params.

Reply:
Possible, this was added in the previous PR. I'll change it

Reply:
@asheshambasta do we actually want to remove parameters, instead of modifying/redacting them as in the original PR?
It seems to me that you don't want to hide the information (that a password param was given) in the logs

Reply:
Yeah, I'd be more in favour of not being able to completely remove param. information either. But I can see this as a  valid use case. For example, say, a browser or a particular redirect causes some business-specific params to be passed: things like `referrer` and `campaign`, are usually just noise that can be handy to remove if at all possible. 
I don't have a strong opinion on this, but maybe the use of `Maybe` still leaves that choice in the hands of the user, should they ever decide to do this?

Reply:
And this leads me to another question: what about filtering headers? They contain sensitive information too. We use:
- the standard `Authorization` header to pass `Bearer` tokens, which must be hidden.
- within services, also the `X-Bearer-Inf` header that contains grant information of the calling user, which also we would like to hide at times.

Reply:
Yeah, noise reduction is a valid use-case. I'll give it a shot. I'll look into headers too.
Once we have functionality, I'll look into upstreaming

Reply:
@asheshambasta Params can be hidden now.
I've looked into headers, but they are not shown by default (except for `Accept`)